csrc/python_frontend/fusion_cache.cpp

```diff
@@ -151,23 +304,39 @@ void FusionCache::stats(std::ostream& os) const {
   }
 }
 
-void FusionCache::reset() {
+void FusionCache::reset(bool load_from_default_workspace) {
   std::lock_guard<std::mutex> guard(singleton_lock_);
   if (singleton_ != nullptr) {
     auto max_fusions = singleton_->max_fusions_;
     delete singleton_;
-    singleton_ = new FusionCache(max_fusions);
+    singleton_ = new FusionCache(max_fusions, load_from_default_workspace);
   }
 }
 
-FusionCache::FusionCache(size_t max_fusions)
+FusionCache::FusionCache(size_t max_fusions, bool load_from_default_workspace)
     : max_fusions_(max_fusions),
       root_(nullptr),
       fusions_(),
       terminal_nodes_(),
       user_def_input_encodings_() {
   RecordFunctor* start = new StartRecord();
   root_ = std::make_unique<TrieNode>(start);
+
+  // Deserialize cache hierarchy from common workspace automatically
+  auto file_path = getSerdeFilePath(getSerdeFile()).native();
+  if (load_from_default_workspace && fs::exists(file_path)) {
+    const BinaryBuffer& buffer = openFusionCache(file_path);
+    const serde::FusionCache* fc =
+        verifyFusionCache(buffer, false /* strict */);
+    // The saved workspace can become out-of-date between nvfuser updates.
+    if (fc != nullptr) {
+      // Only deserialize if the current binary is valid.
+      deserialize(buffer, fc);
+    } else {
+      // Delete incompatible workspace.
+      fs::remove(file_path);
```

Comment:
nitpick: maybe throw a warning here when we are removing things.
I'm also wondering, should this be put into a try block? e.g. if we try to remove something that we can't write to.